compiler/src/iree/compiler/Codegen/Common/test/generic_vectorization.mlir

```diff
@@ -408,6 +408,33 @@ func.func @dynamic_fill_with_scalable_tiling_infer_vector_size(%arg0: tensor<1x6
 
 // -----
 
+
+func.func @pad_lowered_as_masked_transfer_read(%arg0: tensor<?x?xf32>, %arg1: index) -> tensor<1x4xf32> {
+  %cst = arith.constant 0.000000e+00 : f32
+  %padded = tensor.pad %arg0 low[0, 0] high[%arg1, 0] {
+    ^bb0(%arg2: index, %arg3: index):
+      tensor.yield %cst : f32
+  } : tensor<?x?xf32> to tensor<1x4xf32>
+  return %padded : tensor<1x4xf32>
```

Comment:
Note that it only works when the output shape is static.

https://github.com/iree-org/iree/blob/ce2585aa5db62151157e9af4b0449109e9335f4c/compiler/src/iree/compiler/Codegen/Common/GenericVectorization.cpp#L65-L70

Reply:
Right, luckily after tiling of the target padded gemm this seem to always become a static shape that allow me to take advantage of the pad op lowering. So I skipped this in the initial round of optimization, just so that I can demonstrate some end to end perf improvement first.

What's your take on how important to come up with a pad op vectorization inference routine? I'd be interested to pick it up as a backlog once this becomes necessary.

Reply:
Well, it is in this state for a while, so I think it is not that important. I've been wanting to design a new vectorization op interface, but I never get a chance. We can chat more if it becomes a need.